hw/xbox/nv2a/nv2a_psh.c

```diff
@@ -49,120 +49,128 @@
-    PS_TEXTUREMODES_DPNDNT_AR=            0x0fL, // - * * *
-    PS_TEXTUREMODES_DPNDNT_GB=            0x10L, // - * * *
-    PS_TEXTUREMODES_DOTPRODUCT=           0x11L, // - * * -
-    PS_TEXTUREMODES_DOT_RFLCT_SPEC_CONST= 0x12L, // - - - *
-    // 0x13-0x1f reserved
+enum PS_TEXTUREMODES { // valid in stage 0 1 2 3
+    PS_TEXTUREMODES_NONE = 0x00L, // * * * *
+    PS_TEXTUREMODES_PROJECT2D = 0x01L, // * * * *
+    PS_TEXTUREMODES_PROJECT3D = 0x02L, // * * * *
+    PS_TEXTUREMODES_CUBEMAP = 0x03L, // * * * *
+    PS_TEXTUREMODES_PASSTHRU = 0x04L, // * * * *
+    PS_TEXTUREMODES_CLIPPLANE = 0x05L, // * * * *
+    PS_TEXTUREMODES_BUMPENVMAP = 0x06L, // - * * *
+    PS_TEXTUREMODES_BUMPENVMAP_LUM = 0x07L, // - * * *
+    PS_TEXTUREMODES_BRDF = 0x08L, // - - * *
+    PS_TEXTUREMODES_DOT_ST = 0x09L, // - - * *
+    PS_TEXTUREMODES_DOT_ZW = 0x0aL, // - - * *
+    PS_TEXTUREMODES_DOT_RFLCT_DIFF = 0x0bL, // - - * -
+    PS_TEXTUREMODES_DOT_RFLCT_SPEC = 0x0cL, // - - - *
+    PS_TEXTUREMODES_DOT_STR_3D = 0x0dL, // - - - *
+    PS_TEXTUREMODES_DOT_STR_CUBE = 0x0eL, // - - - *
+    PS_TEXTUREMODES_DPNDNT_AR = 0x0fL, // - * * *
+    PS_TEXTUREMODES_DPNDNT_GB = 0x10L, // - * * *
+    PS_TEXTUREMODES_DOTPRODUCT = 0x11L, // - * * -
+    PS_TEXTUREMODES_DOT_RFLCT_SPEC_CONST = 0x12L, // - - - *
+                                                  // 0x13-0x1f reserved
 };
 
-enum PS_INPUTMAPPING
-{
-    PS_INPUTMAPPING_UNSIGNED_IDENTITY= 0x00L, // max(0,x)         OK for final combiner
-    PS_INPUTMAPPING_UNSIGNED_INVERT=   0x20L, // 1 - max(0,x)     OK for final combiner
-    PS_INPUTMAPPING_EXPAND_NORMAL=     0x40L, // 2*max(0,x) - 1   invalid for final combiner
-    PS_INPUTMAPPING_EXPAND_NEGATE=     0x60L, // 1 - 2*max(0,x)   invalid for final combiner
-    PS_INPUTMAPPING_HALFBIAS_NORMAL=   0x80L, // max(0,x) - 1/2   invalid for final combiner
-    PS_INPUTMAPPING_HALFBIAS_NEGATE=   0xa0L, // 1/2 - max(0,x)   invalid for final combiner
-    PS_INPUTMAPPING_SIGNED_IDENTITY=   0xc0L, // x                invalid for final combiner
-    PS_INPUTMAPPING_SIGNED_NEGATE=     0xe0L, // -x               invalid for final combiner
+enum PS_INPUTMAPPING {
+    PS_INPUTMAPPING_UNSIGNED_IDENTITY =
+        0x00L, // max(0,x)         OK for final combiner
+    PS_INPUTMAPPING_UNSIGNED_INVERT =
+        0x20L, // 1 - max(0,x)     OK for final combiner
+    PS_INPUTMAPPING_EXPAND_NORMAL =
+        0x40L, // 2*max(0,x) - 1   invalid for final combiner
+    PS_INPUTMAPPING_EXPAND_NEGATE =
+        0x60L, // 1 - 2*max(0,x)   invalid for final combiner
+    PS_INPUTMAPPING_HALFBIAS_NORMAL =
+        0x80L, // max(0,x) - 1/2   invalid for final combiner
+    PS_INPUTMAPPING_HALFBIAS_NEGATE =
+        0xa0L, // 1/2 - max(0,x)   invalid for final combiner
+    PS_INPUTMAPPING_SIGNED_IDENTITY =
+        0xc0L, // x                invalid for final combiner
+    PS_INPUTMAPPING_SIGNED_NEGATE =
+        0xe0L, // -x               invalid for final combiner
```

Comment:
Formatting of this entire bog is bad in my opinion. The table format was much better (even if it exceeds 80 cols).